roles/reset_proxmox_lxc/tasks/main.yml

```
@@ -0,0 +1,53 @@
+---
+- name: check for container files that exist on this host
+  stat:
+    path: "/etc/pve/lxc/{{ item }}.conf"
+  loop: "{{ proxmox_lxc_ct_ids }}"
+  register: stat_results
+
+- name: filter out files that do not exist
+  set_fact:
+    proxmox_lxc_filtered_files:
+      '{{ stat_results.results | rejectattr("stat.exists", "false") | map(attribute="stat.path") }}'
+
+# used for the reboot handler
+- name: get container ids from filtered files
+  set_fact:
+    proxmox_lxc_filtered_ids:
+      '{{ proxmox_lxc_filtered_files | map("split", "/") | map("last") | map("split", ".") | map("first") }}'
+
```

for the sake of shorter logs for users: Can we merge this into a single task?

---

No, both of these variables are used.

Also, these filters are already pretty relatively complicated, I think the readability of having them separate is worth having one extra task, even if the variable wasn't needed.

---

> No, both of these variables are used.

Obviously :-) What are you getting at?

> readability of having them separate

What do you think of this?

```suggestion
- name: Build list of files and container IDs
  vars:
    existing_files: >
      {{ stat_results.results | rejectattr("stat.exists", "false") | map(attribute="stat.path") }}
  set_fact:
    proxmox_lxc_filtered_files: >
      {{ existing_files }}
    proxmox_lxc_filtered_ids: >
      {{ existing_files | map("split", "/") | map("last") | map("split", ".") | map("first") }}
```

same logic, no bigger filter expressions than before, still more compact to developers as well as users of the playbook 👍🏻

---

Oh, think I just misunderstood, gotcha yeah, indifferent here. Don't need that existing_files var though no? could just have that  set directly on proxmox_lxc_filtered_files and then use that in proxmox_lxc_filtered_ids

---

> Don't need that existing_files var though no?

I'm pretty sure that the variable is needed (since this is not imperative programming, and as a coarse model, one could say that both of the facts are set at the same time). For example, this playbook cannot be run:

```yaml
- hosts: localhost
  tasks:
    - ansible.builtin.set_fact:  # You will encounter an error while executing this task
        foo: "42"
        bar: "{{ foo }}"

    - ansible.builtin.debug:
        msg: foo={{foo}}, bar={{bar}}
```

---

gotcha, alright, if we do this I think it would be worth it to update the implementation as well, rather than just the reset task. Probably best as its own PR, this one has been open for quite a while now.

---

Feel free to adapt the implementation as well, I'm all for it 👍🏻
As for this specific snippet: If we agree that this is the pattern we should use, the patch is ready and one would just click a button to apply it, I see no reason to merge it with the old pattern - do you?

---

I'll leave that up to @Nomsplease and if they want to do another revision and round of testing to make sure there is nothing we haven't considered :)

---

I like this idea, but I feel like we are getting a little into the weeds on this. 

The current system functions and I think is more clear to a newer Anisble user, and we have already done all the validation testing on the massive amount of changes in the PR.

I would say if this is still desired to open another PR to adjust these filter functions, I do not intended to implement this change.

---

It is more than CI, this role isn't covered by the molecule tests (for reasons that can be seen in this [PR](https://github.com/techno-tim/k3s-ansible/pull/209)), you have to manually create/destroy a new cluster with lxc containers to test this change.